SimulationAgent/SimulationContext.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Threading.Tasks;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Concurrency;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.DataStructures;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Runtime;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.SimulationAgent
+{
+    public interface ISimulationContext
+    {
+        IRateLimiting RateLimiting { get; }
+        IDevices Devices { get; }
+        ConnectionLoopSettings ConnectionLoopSettings { get; }
+        PropertiesLoopSettings PropertiesLoopSettings { get; }
+
+        Task InitAsync(Simulation simulation);
+
+        // Invoked by SimulationManager.NewConnectionLoop()
```

Comment:
I'm going to leave this code in and address it in a follow-up PR. This code is related to a perf optimization that was added at some point, but appears not to be, strictly speaking, required. I want to leave the comments in so that I'll remember to add it back.